logging/telemetryhook.go

```diff
@@ -48,7 +48,25 @@ func createAsyncHookLevels(wrappedHook logrus.Hook, channelDepth uint, maxQueueD
 	}
 
 	go func() {
-		defer hook.wg.Done()
+		defer func() {
+			// flush the channel
+			moreEntries := true
+			for moreEntries {
+				select {
+				case entry := <-hook.entries:
+					hook.appendEntry(entry)
```

Comment:
I don't think you need to call `appendEntry`, call `hook.wg.Done()` and skip the range below

Reply:
At this point, there may or may not be elements in hook.pending. 
For every element in hook.entries and hook.pending, there should be a hook.wg.Done()
So, there needs to be two range loops to achieve this.